src/components/ShareButtons/ShareButtonGroup.tsx

```diff
@@ -10,36 +10,60 @@ import CopyLinkButton from './CopyLinkButton';
 import { ShareButton, SocialButtonsContainer } from './ShareButtons.style';
 
 const ShareImageButtons: React.FC<{
-  imageUrl: string;
+  imageUrl: string | (() => Promise<string>);
   imageFilename: string;
-  url: string;
+  url: string | (() => Promise<string>);
   quote: string;
   hashtags: string[];
 }> = ({ imageUrl, imageFilename, url, quote, hashtags }) => {
-  const [showSocialButtons, setShowSocialButtons] = useState(false);
-  const hideSocialButtons = () => {
-    const timeoutId = setTimeout(() => setShowSocialButtons(false), 1500);
+  // Turn url / imageUrl into asynchronous getters if they aren't already.
+  const getUrl = typeof url === 'string' ? () => Promise.resolve(url) : url;
+  const getImageUrl =
+    typeof imageUrl === 'string' ? () => Promise.resolve(imageUrl) : imageUrl;
+
+  const [socialSharingProps, setSocialSharingProps] = useState<{
+    url: string;
+    quote: string;
+    socialIconSize: number;
+  } | null>(null);
```

Comment:
Since we don't synchronously have the URL anymore I wrapped this up in a state variable that we set once we asynchronously generate the URL after the user clicks the share button.